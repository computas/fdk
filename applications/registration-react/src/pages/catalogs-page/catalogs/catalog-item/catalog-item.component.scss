@import '../../../../../node_modules/designsystemet/fdk-designsystem-bootstrap4/scss/colors.scss';
@import '../../../../../node_modules/designsystemet/fdk-designsystem-bootstrap4/scss/mixins.scss';

.catalog-item {
  color: $color-dark-1;
  text-decoration: none;
  &:hover, &:focus, &:active {

  }
}

.catalog-item__body {
  background-color: $color_white;
  border-radius: 5px;
  padding: 2em;
  word-wrap: break-word;
  @include fdk-shadow--regular;
  @include fdk-grow--regular;
  &:focus, &:active {
    box-shadow: none;
    transform: none;
  }
}

.catalog-icon {
  background-repeat: no-repeat;
  background-position: center 0;
  background-size: 2.5em;
  padding-top: 3em;
}

.catalog-icon--dataset {
  background-image: url("/static/img/icon-catalog-dataset.svg");
}

.catalog-icon--api {
  background-image: url("/static/img/icon-catalog-api.svg");
}

.catalog-icon--begrep {
  background-image: url("/static/img/icon-catalog-begrep.svg");
}

.catalog-icon--protokoll {
  background-image: url("/static/img/icon-catalog-protokoll.svg");
}