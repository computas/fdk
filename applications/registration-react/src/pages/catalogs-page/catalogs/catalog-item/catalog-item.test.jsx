import React from 'react';
import { shallow } from 'enzyme';
import { CatalogItem } from './catalog-item.component';

let defaultProps;
let wrapper;

beforeEach(() => {
  defaultProps = {
    publisherId: '123',
    type: 'datasets'
  };
  wrapper = shallow(<CatalogItem {...defaultProps} />);
});

test('should render CatalogItem correctly with missing itemsCount', () => {
  expect(wrapper).toMatchSnapshot();
});

test('should render CatalogItem correctly with itemsCount set to 10', () => {
  wrapper.setProps({
    itemsCount: 10
  });
  expect(wrapper).toMatchSnapshot();
});
