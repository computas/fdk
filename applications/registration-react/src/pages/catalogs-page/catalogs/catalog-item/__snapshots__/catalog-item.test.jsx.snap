// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render CatalogItem correctly with itemsCount set to 10 1`] = `
<div
  className="col-md-4 pl-0"
>
  <Link
    className="catalog-item"
    replace={false}
    to="/catalogs/123/datasets"
  >
    <div
      className="catalog-item__body d-flex flex-column align-items-center"
    >
      <h3
        className="catalog-icon catalog-icon--dataset"
      >
        Datasettkatalog
      </h3>
      <span
        className="fdk-text-size-small fdk-color-dark-2"
      >
        10
         
        datasett
      </span>
    </div>
  </Link>
</div>
`;

exports[`should render CatalogItem correctly with missing itemsCount 1`] = `
<div
  className="col-md-4 pl-0"
>
  <Link
    className="catalog-item"
    replace={false}
    to="/catalogs/123/datasets"
  >
    <div
      className="catalog-item__body d-flex flex-column align-items-center"
    >
      <h3
        className="catalog-icon catalog-icon--dataset"
      >
        Datasettkatalog
      </h3>
    </div>
  </Link>
</div>
`;
